### PR TITLE
Try to avoid StopIteration error when no sql capabilities are available.

### LINKED ISSUE #6450: az sql db list-editions doesn't work @ shell.azure.com
**Describe the bug**
If I run **az sql db list-editions --location "Australia East"** at shell.azure.com it throws exception.
Works from local PowerShell.

**Stack trace is this:**
Traceback (most recent call last):
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\knack\cli.py", line 197, in invoke
    cmd_result = self.invocation.execute(args)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\core\commands\__init__.py", line 336, in execute
    six.reraise(*sys.exc_info())
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\six.py", line 693, in reraise
    raise value
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\core\commands\__init__.py", line 310, in execute
    result = cmd(params)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\core\commands\__init__.py", line 171, in __call__
    return super(AzCliCommand, self).__call__(*args, **kwargs)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\knack\commands.py", line 109, in __call__
    return self.handler(*args, **kwargs)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\core\__init__.py", line 420, in default_command_handler
    result = op(**command_args)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\command_modules\sql\custom.py", line 729, in db_list_capabilities
    editions = _get_default_capability(capabilities.supported_server_versions).supported_editions
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\command_modules\sql\custom.py", line 77, in _get_default_capability
    next(c for c in capabilities if c.status == CapabilityStatus.available))
StopIteration

**Expected behavior**
To work same as on local machine in PowerShell = return list of SQL DB editions info.

**Environment summary**
https://shell.azure.com in Chrome

**Additional context**
Tried for Australia and Europe. Tried with short name and display name of location. Doesn't work with either.

### PR DESCRIPTION
**Pull request by jaredmoo**
Fixes #6450.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py [hunk 1]
```diff
@@ -71,10 +71,14 @@ def _get_default_capability(capabilities):
     '''
     Gets the first capability in the collection that has 'default' status.
     If none have 'default' status, gets the first capability that has 'available' status.
+    If still none are available, just gets the first one, which (by elimination) must
+    be unavailable, which at least ensures that the user does not get a confusing StopIteration
+    error.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py]: Why not just catch the StopIteration and throw an error like "No edition is available to your subscription"?

Comment (jaredmoo) [reply src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py]: That would be simpler, but previous versions (2.0.22 and earlier) would successfully return all the editions. They would just be shown as being unavailable.

Comment (jaredmoo) [reply src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py]: Actually, let me implement this a little bit differently. Will send update in a minute.

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py [hunk 2]
```diff
@@ -81,6 +81,29 @@ def _any_sku_values_specified(sku):
...
+    '''
+    server_versions = location_capabilities.supported_server_versions
+
+    try:
+        # Try behavior from azure-cli-sql 2.0.26: get default
+        return _get_default_capability(server_versions)
+    except StopIteration:
+        # No default or available version found.
+        # Fall back to behaviour from azure-cli-sql 2.0.25 and earlier: get version 12.0
+        return next(sv for sv in server_versions if sv.name == "12.0")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py]: You might issue a warning in this case if it's abnormal.

Comment (jaredmoo) [reply src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py]: How would I do that?

**Comment by jaredmoo** [comment]
Just waiting for merge here. The warning isn't really needed, and if desired it can be added later

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py (hunk 2 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py]: logger.warning(…)

**Pull request merged**